subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/BuildTreeConfigurationCache.kt

```diff
@@ -33,8 +33,10 @@ interface BuildTreeConfigurationCache {
     /**
      * Loads the scheduled tasks from cache, if available, or else runs the given function to schedule the tasks and then
      * writes the result to the cache.
+     *
+     * @param tasksOnly True if tasks execution only was requested. False if model building for tasks was requested.
      */
-    fun loadOrScheduleRequestedTasks(graph: BuildTreeWorkGraph, graphBuilder: BuildTreeWorkGraphBuilder?, scheduler: (BuildTreeWorkGraph) -> BuildTreeWorkGraph.FinalizedGraph): WorkGraphResult
+    fun loadOrScheduleRequestedTasks(graph: BuildTreeWorkGraph, graphBuilder: BuildTreeWorkGraphBuilder?, scheduler: (BuildTreeWorkGraph) -> BuildTreeWorkGraph.FinalizedGraph, tasksOnly: Boolean): WorkGraphResult
```

Comment:
IIUC, `taskExecutionOnly` seems a more accurate name based on the provided description than `tasksOnly` (we are requesting task execution only, as opposed to task execution *and* model building, but it is always about tasks).

Reply:
@abstratt @h0tk3y Great comments, I'll try to choose a new name for the parameter.
I wouldn't want to add _execution_ word since this code path is operating over few verbs : _load_, _schedule_, _run_ for tasks. So adding _execution_ here can be confusing a bit